### PR TITLE
Tiles reorganisation

### PR DESCRIPTION
**Pull request by pilz**
This is a Pull Request which is not intended to be merged but to help to understand how the new Jenkins PR test module is actually behaving. Feel free to review code but PLEASE DO NOT MERGE YET!

### CODE REVIEW

File: src/ploneintranet/workspace/tests/test_content_type.py [hunk 1]
```diff
@@ -186,3 +186,32 @@ def test_cannot_add_other_content_in_workspacecontainer(self):
...
+            'workspaces',
+            title='Workspaces'
+        )
+        ws = api.content.create(
+            wsc,
+            'ploneintranet.workspace.workspacefolder',
+            'workspace-1',
+            title='Workspace 1'
+        )
+        self.assertEquals(ws.workspace_visible, True)
```
Comment (adamcheasley) [inline src/ploneintranet/workspace/tests/test_content_type.py]: We should use assertEqual here, as assertEquals is deprecated:
https://docs.python.org/2/library/unittest.html#deprecated-aliases

Comment (pilz) [inline src/ploneintranet/workspace/tests/test_content_type.py]: Thanks, done that.

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	                <p id="help-partpol-4" class="pat-message notice pat-depends" data-pat-depends="condition: partpol=4">
+	                    <strong>Moderate:</strong> Workspace members can do everything: read, add, publish and edit content.
+	                </p>
+	                <p>
+	                    The participant policy is the default permission settings for all members of this workspace. Execptions for individual people may be made on the <a href="/feedback/workspace-settings-members.html" class="pat-inject" data-pat-inject="target: #workspace-settings; source: #workspace-settings">Members tab</a>.
+	                </p>
+	            </fieldset>
+		    </form>
+		</div>
+	</div>
```
Comment (adrianwhite) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: Would it be possible to use space/soft-tabs rather than hard-tabs for indentation.

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+</div>
+</body>
+</html>
```
Comment (adrianwhite) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: Please could there be a newline at the end of all files.

Comment (pilz) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: Is there a way to have this checked for html files automatically as well like we do with flake for python?

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt (hunk 2 above)
Comment (pilz) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: Yes, done. This comes from jekyll, so it is bound to happen again. Might be interesting to see if we can get jekyll to not use tabs.

Comment (adrianwhite) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: Ah yes, I forgot this was from the generated files. I'll look into it.

**Comment by adamcheasley** [comment]
I know this is controversial, but I would be tempted to squash some of these commits. There's quite a few of them, some of which are not very descriptive. There's also a duplicate in there. Currently, this is going to make the git log quite hard to follow.

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt (hunk 3 above)
Comment (adrianwhite) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: I'm no Jenkins expert but could we do something along these lines:
http://www.phase2technology.com/blog/having-the-butler-tidy-up-your-html/

**Comment by pilz** [comment]
Yes, I agree. I am cleaning up a lot at the moment. We cannot merge automatically to master either because too much refactoring has happened on the way. But it is a lot of work. You are very welcome to squash if you find obvious things, I will continue to clean up until the last change here has been made so that we can merge this before next week.

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt (hunk 2 above)
Comment (pilz) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: I hope I got them all now.
> On 4 Mar 2015, at 17:38, Adrian White <notifications@github.com> wrote:
> 
> In src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt <https://github.com/ploneintranet/ploneintranet.workspace/pull/59#discussion_r25789156>:
> 
> > +	                <p id="help-partpol-2" class="pat-message notice pat-depends" data-pat-depends="condition: partpol=2">
> > +	                    <strong>Produce:</strong> Workspace members can read and add content. They can neither publish nor edit content.
> > +	                </p>
> > +	                <p id="help-partpol-3" class="pat-message notice pat-depends" data-pat-depends="condition: partpol=3">
> > +	                    <strong>Publish:</strong> Workspace members can read, add and publish content. They cannot edit other member's content.
> > +	                </p>
> > +	                <p id="help-partpol-4" class="pat-message notice pat-depends" data-pat-depends="condition: partpol=4">
> > +	                    <strong>Moderate:</strong> Workspace members can do everything: read, add, publish and edit content.
> > +	                </p>
> > +	                <p>
> > +	                    The participant policy is the default permission settings for all members of this workspace. Execptions for individual people may be made on the <a href="/feedback/workspace-settings-members.html" class="pat-inject" data-pat-inject="target: #workspace-settings; source: #workspace-settings">Members tab</a>.
> > +	                </p>
> > +	            </fieldset>
> > +		    </form>
> > +		</div>
> > +	</div>
> Ah yes, I forgot this was from the generated files. I'll look into it.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ploneintranet/ploneintranet.workspace/pull/59/files#r25789156>.
> 

--
Alexander Pilz
SYSLAB.COM GmbH, Landwehrstrasse 60-62, 80336 Munich, Germany
Tel: +49 89 3063589-0 - Fax: +49 89 3063589-9
http://www.syslab.com - Amtsgericht Muenchen, HRB 135057
Steuernummer 143/184/50154 - Ust.-ID: DE212842815

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt (hunk 3 above)
Comment (pilz) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: Nice, but tidy can also make quite a mess. I wouldn’t want to run it without supervision :)

> On 4 Mar 2015, at 17:45, Adrian White <notifications@github.com> wrote:
> 
> In src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt <https://github.com/ploneintranet/ploneintranet.workspace/pull/59#discussion_r25789822>:
> 
> > +	                <p id="help-partpol-3" class="pat-message notice pat-depends" data-pat-depends="condition: partpol=3">
> > +	                    <strong>Publish:</strong> Workspace members can read, add and publish content. They cannot edit other member's content.
> > +	                </p>
> > +	                <p id="help-partpol-4" class="pat-message notice pat-depends" data-pat-depends="condition: partpol=4">
> > +	                    <strong>Moderate:</strong> Workspace members can do everything: read, add, publish and edit content.
> > +	                </p>
> > +	                <p>
> > +	                    The participant policy is the default permission settings for all members of this workspace. Execptions for individual people may be made on the <a href="/feedback/workspace-settings-members.html" class="pat-inject" data-pat-inject="target: #workspace-settings; source: #workspace-settings">Members tab</a>.
> > +	                </p>
> > +	            </fieldset>
> > +		    </form>
> > +		</div>
> > +	</div>
> > +</div>
> > +</body>
> > +</html>
> I'm no Jenkins expert but could we do something along these lines:
> http://www.phase2technology.com/blog/having-the-butler-tidy-up-your-html/ <http://www.phase2technology.com/blog/having-the-butler-tidy-up-your-html/>
> —
> Reply to this email directly or view it on GitHub <https://github.com/ploneintranet/ploneintranet.workspace/pull/59/files#r25789822>.
> 

--
Alexander Pilz
SYSLAB.COM GmbH, Landwehrstrasse 60-62, 80336 Munich, Germany
Tel: +49 89 3063589-0 - Fax: +49 89 3063589-9
http://www.syslab.com - Amtsgericht Muenchen, HRB 135057
Steuernummer 143/184/50154 - Ust.-ID: DE212842815

Comment (pilz) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt]: Sorry, just as a notifier? That would be great!

> On 4 Mar 2015, at 17:47, Alexander Pilz <pilz@syslab.com> wrote:
> 
> Nice, but tidy can also make quite a mess. I wouldn’t want to run it without supervision :)
> 
>> On 4 Mar 2015, at 17:45, Adrian White <notifications@github.com <mailto:notifications@github.com>> wrote:
>> 
>> In src/ploneintranet/workspace/browser/tiles/templates/sidebar-settings-security.pt <https://github.com/ploneintranet/ploneintranet.workspace/pull/59#discussion_r25789822>:
>> 
>> > +	                <p id="help-partpol-3" class="pat-message notice pat-depends" data-pat-depends="condition: partpol=3">
>> > +	                    <strong>Publish:</strong> Workspace members can read, add and publish content. They cannot edit other member's content.
>> > +	                </p>
>> > +	                <p id="help-partpol-4" class="pat-message notice pat-depends" data-pat-depends="condition: partpol=4">
>> > +	                    <strong>Moderate:</strong> Workspace members can do everything: read, add, publish and edit content.
>> > +	                </p>
>> > +	                <p>
>> > +	                    The participant policy is the default permission settings for all members of this workspace. Execptions for individual people may be made on the <a href="/feedback/workspace-settings-members.html" class="pat-inject" data-pat-inject="target: #workspace-settings; source: #workspace-settings">Members tab</a>.
>> > +	                </p>
>> > +	            </fieldset>
>> > +		    </form>
>> > +		</div>
>> > +	</div>
>> > +</div>
>> > +</body>
>> > +</html>
>> I'm no Jenkins expert but could we do something along these lines:
>> http://www.phase2technology.com/blog/having-the-butler-tidy-up-your-html/ <http://www.phase2technology.com/blog/having-the-butler-tidy-up-your-html/>
>> —
>> Reply to this email directly or view it on GitHub <https://github.com/ploneintranet/ploneintranet.workspace/pull/59/files#r25789822>.
>> 
> 
> --
> Alexander Pilz
> SYSLAB.COM <http://syslab.com/> GmbH, Landwehrstrasse 60-62, 80336 Munich, Germany
> Tel: +49 89 3063589-0 - Fax: +49 89 3063589-9
> http://www.syslab.com <http://www.syslab.com/> - Amtsgericht Muenchen, HRB 135057
> Steuernummer 143/184/50154 - Ust.-ID: DE212842815
> 

--
Alexander Pilz
SYSLAB.COM GmbH, Landwehrstrasse 60-62, 80336 Munich, Germany
Tel: +49 89 3063589-0 - Fax: +49 89 3063589-9
http://www.syslab.com - Amtsgericht Muenchen, HRB 135057
Steuernummer 143/184/50154 - Ust.-ID: DE212842815

**Comment by adamcheasley** [comment]
Unfortunately, I think if I try to squash from this branch we would run into problems. Rewriting history on a shared branch can be tricky. If it's just you working on this branch you should be able to do it without issue.

**Comment by pilz** [comment]
we are four people writing on that branch atm
> On 4 Mar 2015, at 17:50, Adam Forsythe-Cheasley <notifications@github.com> wrote:
> 
> Unfortunately, I think if I try to squash from this branch we would run into problems. Rewriting history on a shared branch can be tricky. If it's just you working on this branch you should be able to do it without issue.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ploneintranet/ploneintranet.workspace/pull/59#issuecomment-77195840>.
> 

--
Alexander Pilz
SYSLAB.COM GmbH, Landwehrstrasse 60-62, 80336 Munich, Germany
Tel: +49 89 3063589-0 - Fax: +49 89 3063589-9
http://www.syslab.com - Amtsgericht Muenchen, HRB 135057
Steuernummer 143/184/50154 - Ust.-ID: DE212842815

**Comment by adamcheasley** [comment]
It would also be great to get an entry in the change log for this.

**Comment by pilz** [comment]
added a changelog line.

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+      <li>
+        <a class="icon-document" href="#">Paste document(s)</a>
+      </li>
+    </ul>
+  </div>
+
+  <div id="sidebar-content" class="pat-stacks">
+    <div id="workspace-documents">
+        <div class="item-selector mode-follow" id="selector">
+<!-- <div class="item-selector mode-select" id="selector"> -->
```
Comment (mattss) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt]: We can ditch commented code, as we have version control.

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+                                        title="Copy">Copy</button>
+                                <button disabled
+                                        type="submit"
+                                        name="batch-function"
+                                        value="paste"
+                                        formaction="/open-market-committee#items"
+                                        class="float-before icon-paste no-label"
+                                        title="Paste">Paste</button>
+                            </span>
+                            <!-- <button type="submit" name="batch-function" value="move" class="float-before icon-move" title="Move">Move</button> -->
```
Comment (mattss) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt]: No need for commented code.

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+                    - type-excel
+                    - type-sound
+                    - type-image
+                    - type-code
+                    - type-rich
+                    - type-pdf
+
+                    -->
+
+                    <!-- XXX -->
```
Comment (mattss) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt]: Can this comment be clarified?

File: src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+                           tal:attributes="href string:${item/url};
+                                           data-pat-inject item/dpi">
+                            <strong class="title" tal:content="item/title">Traces through Time data workshop</strong>
+
+                                <dfn class="byline" tal:content="item/description" tal:condition="item/description">An interactive workshop to test new and innovative tools to trace people through time.</dfn>
+
+                        </a>
+                    </label>
+                    </tal:block>
+                    <!-- Notice that there is a dynamic class 'has-description/has no description which depends on whether a description of the object is available' -->
```
Comment (mattss) [inline src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt]: Is any of this commented code needed?

**Comment by mattss** [comment]
Good to go. Waiting for related PR's in theme and suite.
